code/game/objects/items/tools/weldingtool.dm

```diff
@@ -75,7 +75,8 @@
 /obj/item/weldingtool/update_overlays()
 	. = ..()
 	if(change_icons)
-		var/ratio = get_fuel() / max_fuel
+		var/realFuelNoClickbait = reagents.get_reagent_amount(/datum/reagent/toxin/plasma)
```

Comment:
This should be moved to proc get_fuel()

https://github.com/tgstation/tgstation/blob/526e006583eee95d809b4062938072f78974ed78/code/game/objects/items/tools/weldingtool.dm#L197-L198

So it will become 

```
 /obj/item/weldingtool/proc/get_fuel() 
 	return reagents.get_reagent_amount(/datum/reagent/fuel) + reagents.get_reagent_amount(/datum/reagent/toxin/plasma)

```
It's incorrect this is only be checked for in the overlays